bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -120,6 +120,9 @@ public T decode(final BsonReader reader, final DecoderContext decoderContext) {
             InstanceCreator<T> instanceCreator = classModel.getInstanceCreator();
             decodeProperties(reader, decoderContext, instanceCreator);
             return instanceCreator.getInstance();
+        } else if (classModel.getType().isInterface() && !classModel.useDiscriminator()) {
+            reader.skipValue();
```

Comment:
I'm not sure I 100% understand this change - Is it for the case where the user has an interface type but doesn't include a discriminator?  I'm surprised we haven't hit that issue before. 

I don't think we can skip the value - in theory you could have a codec that handles the interface and all implementations of it and returns the correct type based on the data in the document.